osbs/api.py

```diff
@@ -682,10 +683,18 @@ def get_build_logs(self, build_id, follow=False, build_json=None, wait_if_missin
         :param follow: bool, fetch logs as they come?
         :param build_json: dict, to save one get-build query
         :param wait_if_missing: bool, if build doesn't exist, wait
-        :return: None, bytes, or iterable of bytes
+        :param decode: bool, whether or not to decode logs as utf-8
+        :return: None, str, bytes, or iterable of bytes
         """
-        return self.os.logs(build_id, follow=follow, build_json=build_json,
+        logs = self.os.logs(build_id, follow=follow, build_json=build_json,
                             wait_if_missing=wait_if_missing)
+        # we can't decode a Generator
+        if decode and follow:
+            warnings.warn("support for decode when using follow has not been implemented")
```

Comment:
It was a little bit more involved than this, because of the possibility we're returned a string instead of a generator, but I think I've got it figured out now.